src/home/pam_systemd_home.c

```diff
@@ -45,14 +45,23 @@ static int parse_argv(
                         else if (please_suspend)
                                 *please_suspend = k;
 
-                } else if ((v = startswith(argv[i], "debug="))) {
+                } else if ((v = startswith(argv[i], "debug"))) {
                         int k;
 
-                        k = parse_boolean(v);
-                        if (k < 0)
-                                pam_syslog(handle, LOG_WARNING, "Failed to parse debug= argument, ignoring: %s", v);
-                        else if (debug)
-                                *debug = k;
+                        if (*v == 0)
+                                /* We're at the end of the string */
+                                *debug = 1;
+
+                        else if ((v = startswith(v, "="))) {
+                                k = parse_boolean(v);
+                                if (k < 0)
+                                        pam_syslog(handle,
+                                                   LOG_WARNING,
+                                                   "Failed to parse debug= argument, ignoring: %s",
+                                                   v);
+                                else if (debug)
+                                        *debug = k;
+                        }
```

Comment:
i think this would be shorter if written like this:

```c
else if (streq(argv[i], "debug")
        *debug = true;
else if ((v = startswith(argv[i], "debug=")) {
        …
```

could you rework?

(src/login/pam_systemd.c does it in a similar fashion)

Reply:
makes sense, done.